ord_schema/scripts/list_dois.py

```diff
@@ -43,11 +43,15 @@ def main(argv):
     filenames = glob.glob(FLAGS.input, recursive=True)
     logging.info('Found %d datasets', len(filenames))
     dois = collections.defaultdict(list)
+    output_filenames = {}
     for filename in filenames:
         logging.info('Checking %s', filename)
         dataset = message_helpers.load_message(filename, dataset_pb2.Dataset)
-        dataset_id = os.path.splitext(os.path.basename(filename))[0]
-        assert dataset.dataset_id == dataset_id
+        dataset_id = os.path.basename(filename).split('.')[0]
```

Comment:
This snippet is possibly worth splitting out as a helper with the explicit test cases you want to cover